bin/tx-prover/src/proxy/mod.rs

```diff
@@ -239,6 +312,27 @@ impl LoadBalancer {
         let workers_count = self.num_workers().await;
         Some(create_workers_updated_response(session, workers_count).await)
     }
+
+    /// Check the health of the workers and returns a list of healthy workers.
+    ///
+    /// Performs a health check on each worker using the gRPC health check protocol. If a worker
+    /// is not healthy, it won't be included in the list of healthy workers.
+    async fn check_workers_health(
```

Comment:
One thing I'm wondering: how would this affect workers which are currently busy processing a task? If the worker is healthy, it won't have an effect. But if it isn't (i.e., the worker has crushed), wouldn't `Self::logging()` method eventually get called (presumably after some timeout) and the worker will be added back to the list of available workers?

Reply:
If the worker is not healthy, `check_worker_health` will remove it from the list of workers, no matter if it is currently unavailable (i.e. is currently processing a request). And when `Self::logging` is called at the end of the process, since this particular worker is not in the list of workers anymore, it won't be added back.